rn/Teacher/ios/Teacher/Submissions/SubmissionFilterPickerViewController.swift

```diff
@@ -0,0 +1,190 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class SubmissionFilterPickerViewController: UIViewController {
+    let tableView = UITableView(frame: .zero, style: .plain)
+    lazy var resetButton = UIBarButtonItem(title: NSLocalizedString("Reset"), style: .plain, target: self, action: #selector(resetFilter))
+    lazy var doneButton = UIBarButtonItem(title: NSLocalizedString("Done"), style: .done, target: self, action: #selector(done))
+
+    var assignmentID = ""
+    var context = Context.currentUser
+    let env = AppEnvironment.shared
+    var filter: GetSubmissions.Filter?
+    var onChange: ([GetSubmissions.Filter]) -> Void = { _ in }
+    var outOfText: String?
+    var sectionIDs: Set<String> = []
+    let staticFilters: [GetSubmissions.Filter?] = [ nil, .late, .notSubmitted, .needsGrading, .graded ]
+
+    lazy var sections = env.subscribe(GetCourseSections(courseID: context.id)) { [weak self] in
+        self?.tableView.reloadData()
+    }
+
+    static func create(
+        context: Context,
+        outOfText: String?,
+        filter: [GetSubmissions.Filter],
+        onChange: @escaping ([GetSubmissions.Filter]) -> Void
+    ) -> SubmissionFilterPickerViewController {
+        let controller = SubmissionFilterPickerViewController()
+        controller.context = context
+        controller.outOfText = outOfText
+        for f in filter {
+            if case .section(let sectionIDs) = f {
+                controller.sectionIDs = sectionIDs
+            } else {
+                controller.filter = f
```

Comment:
So this will get overridden by the last filter in the filter array. I'm trying to wrap my head around why/how we are going from a list of filters to a single filter.

Reply:
The filters are mutually exclusive, except the sections. So most places will handle an array of filters (`and`ed), but in here we know that the array can have at most 2 items, a section list & something else.

Reply:
I thought about making the section id list something separate, but it seemed easier for most places to just deal with an array of filters, since I didn't want to make `.section()` recursive & somehow exclude its own case.